### ISSUE TITLE
Cannot get 0.99 on MNIST

### ISSUE THREAD

**Issue by ilkarman** [facebookarchive/caffe2#720]
I loosely followed the[ MNIST tutorial](https://github.com/caffe2/caffe2/blob/master/caffe2/python/tutorials/MNIST.ipynb), however use a generator function instead of leveldb IO and try to use higher-level API calls e.g. "optimizer":

https://github.com/ilkarman/NLP_StackOverFlow/blob/master/notebooks/02_Caffe2_CNN.ipynb

E.g.

```
        xent = model.LabelCrossEntropy([softmax, "label"], 'xent')
        loss = model.AveragedLoss(xent, "loss")
        brew.accuracy(model, [softmax, "label"], "accuracy")
        model.AddGradientOperators([loss])
        opt = optimizer.build_sgd(model, base_learning_rate=0.01)
```

However, with [this ](https://github.com/ilkarman/NLP_StackOverFlow/blob/master/notebooks/02_Caffe2_CNN.ipynb)notebook I can't replicate the 0.99 accuracy you would expect from lenet-5 and get from other frameworks.
